tcp_logger.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"io"
+	"log"
+	"net"
+	"os"
+	"time"
+)
+
+// TCPLogger is a logging service that will spool messages in a local channel
+// that are intended to be flushed out to a net.Conn given.
+// Note that messages are not guaranteed to be delivered, but attempts are made to
+// redeliver messages given connection.Write errors.
+type TCPLogger struct {
+	conn          net.Conn
+	deadlineWait  time.Duration
+	retryLimit    int
+	retries       int
+	sleepDuration time.Duration
+
+	logLines chan []byte
+}
+
+// NewTCPLogger is used to establish a new TCPLogger.
+func NewTCPLogger(conn net.Conn, retryLimit int) TCPLogger {
+	return TCPLogger{
+		conn:          conn,
+		deadlineWait:  time.Minute,
+		retryLimit:    retryLimit,
+		sleepDuration: time.Minute,
+
+		logLines: make(chan []byte, 1000), // Should this be a buffered channel?
```

Comment:
I think so. Otherwise we risk blocking the application.

Reply:
👍  This was more of a relic of my thought process when writing, I just removed the comment.